daiquiri/core/management/commands/find_users.py

```diff
@@ -0,0 +1,142 @@
+        parser.add_argument(
+            "--profile_pending",
+            default=".*",
+            help="filter users by profile_pending, requires argument, supports: true/false/1/0",
+        )
+        parser.add_argument(
+            "--profile_confirmed",
+            default=".*",
+            help="filter users by profile_confirmed, requires argument, supports: true/false/1/0",
+        )
+        parser.add_argument(
+            "-p",
+            "--print",
+            action="store_true",
+            help="print found users, don't save them to csv",
+        )
+        parser.add_argument(
+            "-o",
+            "--output_file",
+            default="found_users.csv",
+            help="output file, default is 'found_users.csv'",
+        )
+
+    def save_csv(self, data, filename):
+        if len(data) > 0:
+            data_file = open(filename, "w")
+            csv_writer = csv.writer(data_file)
+            csv_writer.writerow(list(data[0].keys()))
+            for user in data:
+                csv_writer.writerow(user.values())
+            print("List written to " + filename)
+
+    def print_file(self, filename):
+        f = open(filename, "r")
+        content = f.read()
+        print(content)
+        f.close()
+
+    def get_profile(self, username):
+        return Profile.objects.get(user__username=username)
+
+    def rx_match(self, regex, s):
+        return bool(re.search(regex, str(s)))
+
+    def make_bool(self, val):
+        if str(val) == "1":
+            return True
+        if str(val).lower() == "true":
+            return True
+        return False
+
+    def bools_are_equal(self, b1, b2):
+        return self.make_bool(b1) == self.make_bool(b2)
+
+    def check_match(self, user, profile, email_verified, options):
+        if self.rx_match(options["id"], user.id) is False:
+            return False
+        if self.rx_match(options["email"], user.email) is False:
+            return False
+        if self.rx_match(options["username"], user.username) is False:
+            return False
+        if self.rx_match(options["last_name"], user.last_name) is False:
+            return False
+        if (
+            self.bools_are_equal(options["profile_pending"], profile.is_pending)
+            is False
+        ):
+            return False
+        if (
+            self.bools_are_equal(options["profile_confirmed"], profile.is_confirmed)
+            is False
+        ):
+            return False
+        if self.bools_are_equal(options["email_verified"], email_verified) is False:
+            return False
+        return True
+
+    def find_users(self, options):
+        found_users = []
+        for _, user in enumerate(User.objects.all().order_by("date_joined")):
+            profile = Profile.objects.get(user__username=user.username)
+            email = EmailAddress.objects.filter(user=user.id, verified=True).first()
+            email_verified = False
+            if email is not None:
```

Comment:
What is the situation when `email is None` ? is that allowed? should we not catch this case as well?

Reply:
`email = None` occurs when user requested account but was not yet logged-in.